### ISSUE TITLE
Which evaluator + loss function is best for CLIP fine-tuning?

### ISSUE THREAD

**Issue by arditobryan** [UKPLab/sentence-transformers#2289]
Hi, 

I am trying to fine-tune CLIP with the following code:

```
import os
from PIL import Image
import pandas as pd

# Sample data for demonstration
data = {
    'img_name': [f'img_{x+1}.jpg' for x in range(3)],
    'description': ['A tall skyscraper', 'a nice kitten', 'a room full of trees']
}

#fine-tuning dataset
train_captions = pd.DataFrame(data)
train_captions['img_name'] = train_captions['img_name'].apply(lambda x : 'img/'+x)
train_captions

#fine-tuning code
from sentence_transformers import SentenceTransformer, InputExample, losses
from torch.utils.data import DataLoader
from PIL import Image

model = SentenceTransformer("sentence-transformers/clip-ViT-B-32")

train_examples = [InputExample(texts=[Image.open(row[1]['img_name']), row[1]['description']]) for row in train_captions.iterrows()]
train_dataloader = DataLoader(train_examples, shuffle=True, batch_size=16)

train_loss = losses.MultipleNegativesRankingLoss(model=model)

#evaluator?

model.fit([(train_dataloader, train_loss)], show_progress_bar=True, epochs=5, output_path='clip_fine-tuned')
```

Is MultipleNegativesRankingLoss correct as train_loss?
What evaluator should I use? Thank you

**Comment by QGavillet** [comment]
Hi,

I am currently in the same situation, asking the same question. Also isn't there solution to feed the model positives and negatives proposition at the same time for fitting ? As with N-Pair or Triplet Loss.
